platform/ui/src/components/ViewportActionCorners/ViewportActionCorners.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import React, { ReactNode } from 'react';
+
+export enum ViewportActionCornersLocations {
+  topLeft,
+  topRight,
+  bottomLeft,
+  bottomRight,
+}
+
+export type ViewportActionCornersProps = {
+  cornerComponents: Record<ViewportActionCornersLocations, Array<ReactNode>>;
+};
+
+const commonClasses = 'pointer-events-auto flex items-center gap-1';
+const classes = {
+  [ViewportActionCornersLocations.topLeft]: classNames(commonClasses, 'absolute top-[4px]'),
+  [ViewportActionCornersLocations.topRight]: classNames(
+    commonClasses,
+    'absolute top-[4px] right-viewport-scrollbar'
+  ),
+  [ViewportActionCornersLocations.bottomLeft]: classNames(commonClasses, 'absolute bottom-[4px]'),
+  [ViewportActionCornersLocations.bottomRight]: classNames(
+    commonClasses,
+    'absolute bottom-[4px] right-viewport-scrollbar'
+  ),
+};
+
+function ViewportActionCorners({ cornerComponents }: ViewportActionCornersProps) {
+  if (!cornerComponents) {
```

Comment:
probably need if (typeof cornerComponents !== 'object'  &&

Reply:
I am not sure why I need what you are suggesting. I am simply checking for null/undefined here.